tests/test_fieldset.py

```diff
@@ -413,3 +413,50 @@ def generate_dataset(xdim, ydim, zdim=1, tdim=1):
 
     pset.execute(AdvectionRK4, dt=1)
     assert pset[0].lon == 4.5 and pset[0].lat == 10
+
+
+def test_fieldset_from_data_gridtypes(xdim=20, ydim=10, zdim=4):
+    """ Simple test for fieldset initialisation from data. """
+    lon = np.linspace(0., 10., xdim, dtype=np.float32)
+    lat = np.linspace(0., 10., ydim, dtype=np.float32)
+    depth = np.linspace(0., 1., zdim, dtype=np.float32)
+    depth_s = np.ones((zdim, ydim, xdim))
+    U = np.ones((zdim, ydim, xdim))
+    V = np.ones((zdim, ydim, xdim))
+    dimensions = {'lat': lat, 'lon': lon, 'depth': depth}
+    data = {'U': np.array(U, dtype=np.float32), 'V': np.array(V, dtype=np.float32)}
+    lonm, latm = np.meshgrid(lon, lat)
+    for k in range(zdim):
+        data['U'][k, :, :] = lonm * (depth[k]+1) + .1
+        depth_s[k, :, :] = depth[k]
+
+    # Rectilinear Z grid
+    fieldset = FieldSet.from_data(data, dimensions, mesh='flat')
+    pset = ParticleSet(fieldset, ScipyParticle, 0, 0)
+    pset.execute(AdvectionRK4, runtime=1, dt=.5)
+    plon = pset[0].lon
+    plat = pset[0].lat
+
+    # Rectilinear S grid
+    dimensions['depth'] = depth_s
+    fieldset = FieldSet.from_data(data, dimensions, mesh='flat')
+    pset = ParticleSet(fieldset, ScipyParticle, 0, 0)
+    pset.execute(AdvectionRK4, runtime=1, dt=.5)
+    plon = pset[0].lon
+    plat = pset[0].lat
```

Comment:
Assert here that particles have actually moved (i.e. that `plon == 1`)? Because if for some reason there is no movement, then all these asserts will go well

Reply:
Done, I check analytical value.